lib/webdriveragent.js

```diff
@@ -370,6 +370,7 @@ class WebDriverAgent {
       });
     }
 
+    await waitForWdaStart();
```

Comment:
I assume this should be happening after we setup proxies.
Also, there is no need to call getStatus twice, we might merge the loop with  `const status = await this.getStatus();`

Reply:
we may also modify getStatus and add an optional timeout arg there, so it waits automatically if a value is provided

Reply:
sounds good